src/tests/ftest/cart/util/daos_sys_logscan.py

```diff
@@ -0,0 +1,553 @@
+            puuid = match.group(1)
+            ver = int(match.group(2))
+            gen = int(match.group(3))
+            op = match.group(4)
+            if not self._is_leader(puuid, rank, pid):
+                return True
+            term = self._cur_term[puuid]
+            if term < 1:
+                self._warn(f"pool {puuid} I don't know what term it is ({term})!", fname, line)
+                return True
+            if gen in self._pools[puuid][term]["maps"][ver]["rb_gens"]:
+                self._warn(f"pool {puuid} term {term} ver {ver} already has gen {gen}", fname, line)
+            self._pools[puuid][term]["maps"][ver]["rb_gens"][gen] = {
+                "op": op,
+                "start_time": datetime,
+                "time": "xx/xx-xx:xx:xx.xx",
+                "started": True,
+                "scanning": False,
+                "pulling": False,
+                "completed": False,
+                "aborted": False,
+                "duration": 0
+            }
+            if self._debug:
+                print(f"{datetime} FOUND rebuild start in term {term}, rb={puuid}/{ver}/{gen}/{op} "
+                      f"rank {rank}\t{host}\tPID {pid}\t{fname}")
+            return True
+        return False
+
+    def _match_ps_rb_status(self, fname, line, pid, rank):
+        # Do not match on new rebuild log format if we found legacy format
+        if not self._check_rb_new_fmt:
+            return False
+        msg, host, datetime = self._get_line_components(line)
+        match = self.re_rebuild_ldr_status.match(msg)
+        if match:
+            # Disable checking for legacy rebuild log format, to save execution time
+            self._check_rb_legacy_fmt = False
+            puuid, ver, gen, op = self._get_rb_components(match)
+            status = match.group(5)
+            dur = int(match.group(6))
+            if not self._is_leader(puuid, rank, pid):
+                return True
+            term = self._cur_term[puuid]
+            if term < 1:
+                self._warn(f"pool {puuid} I don't know what term it is ({term})!", fname, line)
+                return True
+            if ver not in self._pools[puuid][term]["maps"]:
+                self._warn(f"pool {puuid} term {term} ver {ver} not in maps - add placeholder",
+                           fname, line)
+                self._pools[puuid][term]["maps"][ver] = {
+                    "carryover": False,
+                    "from_ver": ver,
+                    "time": "xx/xx-xx:xx:xx.xx",
+                    "rb_gens": {}
+                }
+                self._pools[puuid][term]["maps"][ver]["rb_gens"][gen] = {
+                    "op": op,
+                    "start_time": "xx/xx-xx:xx:xx.xx",
+                    "time": "xx/xx-xx:xx:xx.xx",
+                    "started": True,
+                    "scanning": False,
+                    "pulling": False,
+                    "completed": False,
+                    "aborted": False,
```

Comment:
Nit, could be simpler to replace these boolean fields with only one status field ?
If needed, an history of the state could also be added to check that it respect the rebuild machine state.